clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/model/RawResource.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2020 Coveo, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.clouddriver.model;
+
+import com.netflix.spinnaker.clouddriver.names.NamerRegistry;
+import com.netflix.spinnaker.moniker.Moniker;
+
+/** A representation of a raw resource. */
+public interface RawResource {
+  /**
+   * Name of the raw resource
+   *
+   * @return name
+   */
+  String getName();
+
+  /**
+   * This resource's moniker
+   *
+   * @return moniker
+   */
+  default Moniker getMoniker() {
```

Comment:
Do we need a default implementation of this (and thus to pull in the `Moniker` dependency)?  So far the only implementation is for Kubernetes which overrides this, so it may be better to just omit the default implementation and only add a default if needed in the future.